### DISCUSSION TITLE
Using a store factory causes state updates

### DISCUSSION THREAD

**Discussion by phaistonian** [pmndrs/zustand#973] [Q&A]
Here's the [sandbox](https://codesandbox.io/s/angry-mclean-47dee0?file=/src/App.js).

In the second scenario,  changing the component causes the local state to be reset.

https://user-images.githubusercontent.com/796530/170499596-95464bce-6ff1-4754-9ef4-46a199254af1.mp4

I am not sure this is related to react-refresh or to Zustand, but it's quite annoying.

Ideas?

**Comment by dai-shi** [comment]
store factory is generally not recommended, because `create` return a hook, which should be used _statically_.

v4 exposes new APIs to create and use vanilla store.
See: https://codesandbox.io/s/goofy-voice-7yjmyh
Does it work as you expect?

**Comment by phaistonian** [comment]
It does work.

However, what if a store is used in both React and Vanilla contexts.

Does wrapping it in the createStore do the trick?

**Reply by dai-shi** [reply]
> what if a store is used in both React and Vanilla contexts.

what does it mean?

**Reply by phaistonian** [reply]
Can I use `createStore` with an existing non-vanilla store?

**Reply by dai-shi** [reply]
No, `createStore` is a base function to `create`, ie. `create` internally uses `createStore`.
